### PR TITLE
New: Adding PAM authentication support to FAB

### PR DESCRIPTION
**Pull request by rabsr**
Reference: #1296 

User will have an option to use Pluggable Authentication Module (PAM) for authentication along with other options available.  By, default 'login' service will be used for PAM authentication. User can override for using any other predefined or custom service.

### CODE REVIEW

**Comment by rabsr** [comment]
@dpgaspar I will test cases.

**Comment by dpgaspar** [comment]
No, got to add it. I'm using the following simple docker-compose:
```
version: '2'
services:
  postgres:
    image: postgres:10
    restart: unless-stopped
    command: postgres -c 'max_connections=500'
    environment:
      POSTGRES_DB: app
      POSTGRES_PASSWORD: pguserpassword
      POSTGRES_USER: pguser
    ports:
      - 5432:5432

  mysql:
    image: mysql:5.7
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: app
      MYSQL_USER: mysqluser
      MYSQL_PASSWORD: mysqluserpassword
      MYSQL_ROOT_PASSWORD: root-password
    networks:
      - default
    ports:
      - 3306:3306

  mssql:
    image: mcr.microsoft.com/mssql/server:2017-CU8-ubuntu
    restart: unless-stopped
    environment:
      ACCEPT_EULA: Y
      SA_PASSWORD: Password_123
    ports:
      - 1433:1433

  mongo:
    image: mongo:4.2.1
    restart: unless-stopped
    ports:
      - 27017:27017
```

then `tox -e postgres`

**Comment by rabsr** [comment]
@dpgaspar I have added added test cases and fixed minor bugs. Please review.

File: flask_appbuilder/templates/appbuilder/general/security/login_pam.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                    <div class="help-block">{{_("Enter your login and password below")}}:</div>
+                    <div class="control-group{% if form.errors.openid is defined %} error{% endif %}">
+                        <label class="control-label" for="openid">{{_("Username")}}:</label>
+
+                        <div class="controls">
+                            <div class="input-group">
+                                <span class="input-group-addon"><i class="fa fa-user"></i></span>
+                                {{ form.username(size = 80, class = "form-control", autofocus = true) }}
+                            </div>
+                            {% for error in form.errors.get('openid', []) %}
```
Comment (mik-laj) [inline flask_appbuilder/templates/appbuilder/general/security/login_pam.html:27]: ?

Comment (rabsr) [reply flask_appbuilder/templates/appbuilder/general/security/login_pam.html:27]: This is here for showing any errors. I have used same login page content as for DB authentication.

**Comment by rabsr** [comment]
@dpgaspar @mik-laj Please review.

**Pull request closed without merging**